spec/controllers/questions_controller_spec.rb

```diff
@@ -75,22 +81,46 @@
 
   describe 'PATCH #update' do
     context 'with valid attributes' do
-      it 'assings the requested question to @question' do
-        patch :update, params: { id: question, question: attributes_for(:question) }
-        expect(assigns(:question)).to eq question
+      context 'owner updating' do
+        it 'assigns the requested question to @question' do
+          patch :update, params: { id: question, question: attributes_for(:question) }
+          expect(assigns(:question)).to eq question
+        end
+
+        it 'changes question attributes' do
+          patch :update, params: { id: question,
+                                   question: attributes_for(:question, title: 'new title', body: 'new body') }
+          question.reload
+          expect(question.title).to eq 'new title'
+          expect(question.body).to eq 'new body'
+        end
+
+        it 'redirects to show view' do
+          patch :update, params: { id: question, question: attributes_for(:question) }
+          expect(request).to redirect_to question_path(assigns(:question))
+        end
       end
 
-      it 'changes question attributes' do
-        patch :update, params: { id: question,
-                                 question: attributes_for(:question, title: 'new title', body: 'new body') }
-        question.reload
-        expect(question.title).to eq 'new title'
-        expect(question.body).to eq 'new body'
+      context 'not owner' do
+        it 'changes question attributes' do
+          sign_in other_user
+          patch :update, params: { id: question,
+                                   question: attributes_for(:question, title: 'new title', body: 'new body') }
+          question.reload
+          expect(question.title).to_not eq 'new title'
+          expect(question.body).to_not eq 'new body'
```

Comment:
change to understand that "other_user" doesn't have the ability to update object at all, not just check values